dask/array/optimization.py

```diff
@@ -41,33 +41,31 @@ def optimize(
     if isinstance(dsk, HighLevelGraph):
         dsk = optimize_blockwise(dsk, keys=keys)
         dsk = fuse_roots(dsk, keys=keys)
+        dsk = dsk.cull(set(keys))
+    dsk = ensure_dict(dsk)
+    dependencies = {k: get_dependencies(dsk, k, as_list=True) for k in dsk}
```

Comment:
We're going to need to be able to get dependencies soon after this I think.  It might be good follow-on work if you're interested.  Also, getting dependencies in blockwise operations without actually having to traverse the graph is likely to be our first actual speed improvement.  It might be worth moving this operation into the HLG section sooner rather than later.  It would feel good to have some performance benefit to show from this rewrite work.

Reply:
Good point, I have added `get_dependencies()` HLG function that does exactly that. 
Updated the PR description to reflect this.